### DISCUSSION TITLE
swmm-toolkit can not install in linux amd64,only have linux_x86_64

### DISCUSSION THREAD

**Discussion by Lijifeng1234** [pyswmm/swmm-python#114] [Q&A]

**Comment by bemcdonnell** [comment]
@Lijifeng1234 you’re going to have to build it.

**Comment by Lijifeng1234** [comment]
> @Lijifeng1234 you’re going to have to build it.
Thank you for your reply !  
I have to build it in linux amd64 now,what should I do ?

**Comment by bemcdonnell** [comment]
@Lijifeng1234, I created a simple guide in this issue (which coincidentally is very similar to your request today).  Give this process a try with amd64 ..   

https://github.com/pyswmm/swmm-python/issues/112#issuecomment-1536209461

**Comment by Lijifeng1234** [comment]
> @Lijifeng1234, I created a simple guide in this issue (which coincidentally is very similar to your request today). Give this process a try with amd64 ..
> 
> [#112 (comment)](https://github.com/pyswmm/swmm-python/issues/112#issuecomment-1536209461)

Thanks,I'll try it now.

**Comment by Lijifeng1234** [comment]
> 

Sorry, the official documentation is as follows

[swmm_toolkit-0.14.0-cp311-cp311-win_amd64.whl ](https://files.pythonhosted.org/packages/49/b3/a4e67a7d1caa67492d2327af4f993d2cccdafc48930563ed3719ff9238d0/swmm_toolkit-0.14.0-cp311-cp311-win_amd64.whl)(383.7 kB [view hashes](https://pypi.org/project/swmm-toolkit/#copy-hash-modal-39581236-fc80-46b0-b060-0bff2dc855e5))
Uploaded Mar 29, 2023 cp311

[swmm_toolkit-0.14.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl ](https://files.pythonhosted.org/packages/45/23/692ab390a11fa0f27dc086679b4e79a39f7b05825a6e15e98c85272722c2/swmm_toolkit-0.14.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl)(487.3 kB [view hashes](https://pypi.org/project/swmm-toolkit/#copy-hash-modal-49e5977d-36e4-40a1-bbe9-1447c0cfb3d4))
Uploaded Mar 29, 2023 cp311

[swmm_toolkit-0.14.0-cp311-cp311-macosx_10_9_x86_64.whl ](https://files.pythonhosted.org/packages/ff/7f/0f887444527cd25851f910e867337814e3b270d6c34cb8c7e79f5ddd0565/swmm_toolkit-0.14.0-cp311-cp311-macosx_10_9_x86_64.whl)(608.2 kB [view hashes](https://pypi.org/project/swmm-toolkit/#copy-hash-modal-6ce1ade6-28ad-4028-9cf9-1f00bd89d606))

just have win_amd64\macosx_x86_64\manylinux_2_17_x86_64, linux_amd64 or linux_aarch64 not exits in range. 

Can you provide the installation files for the above linux version or thr source code compiled files in Linux?

**Comment by bemcdonnell** [comment]
@Lijifeng1234, I am not sure what you were asking for. Cmake is our build generator. All of the build files relevant to the code are all in the repository. Are you asking for the SWMM source code? If so, that is part of a sub module, that you would have to check out with the instructions that I gave in the document

**Comment by Lijifeng1234** [comment]
> @Lijifeng1234, I am not sure what you were asking for. Cmake is our build generator. All of the build files relevant to the code are all in the repository. Are you asking for the SWMM source code? If so, that is part of a sub module, that you would have to check out with the instructions that I gave in the document

Sorry,I got the system architecture wrong initially. I want to install pyswmm and swmm-toolkit in python 3.9 by arrch64_linux server,but pypi.org.com can not provide swmm-toolkit package with arrch64_linux.Meanwhile,installing swmm-toolkit by compiling source code always failed.I can't find the source code for pyswmm completion that compiles successfully in linux_aarch64.Could you help me?

**Comment by bemcdonnell** [comment]
@Lijifeng1234, can you describe what failed when you were trying to build the SWMM-toolkit wheel?  I outlined the whole process in the PDF.  “Pyswmm” is all python code, this repository is the interfacing maps to the pyswmm/stormwater-management-model repository (where the code for SWMM is). If you follow the document line for line, can you tell me what errors you have? 

[build_swmm-toolkit_wheel_v1_20230505.pdf](https://github.com/pyswmm/swmm-python/files/11411912/build_swmm-toolkit_wheel_v1_20230505.pdf)

**Comment by Lijifeng1234** [comment]
> @Lijifeng1234, can you describe what failed when you were trying to build the SWMM-toolkit wheel? I outlined the whole process in the PDF. “Pyswmm” is all python code, this repository is the interfacing maps to the pyswmm/stormwater-management-model repository (where the code for SWMM is). If you follow the document line for line, can you tell me what errors you have?
> 
> [build_swmm-toolkit_wheel_v1_20230505.pdf](https://github.com/pyswmm/swmm-python/files/11411912/build_swmm-toolkit_wheel_v1_20230505.pdf)

python3 setup.py bdist_wheel


CMake Error at swmm-solver/CMakeLists.txt:121 (install):
  install EXPORT given no DESTINATION!


CMake Error at swmm-solver/CMakeLists.txt:130 (install):
  install FILES given no DESTINATION!


install(
    EXPORT
        swmm5Targets
    DESTINATION
        "${CONFIG_DIST}"
    FILE
        swmm5-config.cmake
)

install(
    FILES 
        ${SWMM_PUBLIC_HEADERS} 
    DESTINATION 
        "${INCLUDE_DIST}"
)

**Comment by bemcdonnell** [comment]
@Lijifeng1234 from which directory are you trying to run that command? Is it in your SWMM-toolkit ?

**Comment by Lijifeng1234** [comment]
> @Lijifeng1234 from which directory are you trying to run that command? Is it in your SWMM-toolkit ?

Yes,I got file from "https://github.com/pyswmm/swmm-python",and add "extern","swmm-solver" from "https://github.com/pyswmm/Stormwater-Management-Model" ["/extern","/src/solver"]

[root@localhost swmm-toolkit]# ls
AUTHORS                 CMakeLists.txt  docs    LICENSE.md  setup.py  src          test-requirements.txt  tools
build-requirements.txt  dist            extern  README.md   _skbuild  swmm-solver  tests
[root@localhost swmm-toolkit]# 
[root@localhost swmm-toolkit]# ls extern/
boost.cmake  openmp.cmake  version.cmake  version.h.in

[root@localhost swmm-toolkit]# ls swmm-solver/
bindings        datetime.h  exfil.c     gage.c      iface.c   input.c     lid.h       mempool.c   project.c    routing.c   street.h    toposort.c
climate.c       dwflow.c    exfil.h     globals.h   include   inputrpt.c  lidproc.c   mempool.h   qualrout.c   runoff.c    subcatch.c  transect.c
CMakeLists.txt  dynwave.c   findroot.c  gwater.c    infil.c   keywords.c  link.c      node.c      rain.c       shape.c     surfqual.c  treatmnt.c
consts.h        enums.h     findroot.h  hash.c      infil.h   keywords.h  macros.h    objects.h   rdii.c       snow.c      swmm5.c     version.h
controls.c      error.c     flowrout.c  hash.h      inflow.c  kinwave.c   massbal.c   odesolve.c  report.c     stats.c     table.c     xsect.c
culvert.c       error.h     forcmain.c  headers.h   inlet.c   landuse.c   mathexpr.c  odesolve.h  Roadmap.txt  statsrpt.c  text.h      xsect.dat
datetime.c      error.txt   funcs.h     hotstart.c  inlet.h   lid.c       mathexpr.h  output.c    roadway.c    street.c    toolkit.c
[root@localhost swmm-toolkit]# 

[root@localhost swmm-toolkit]# mkdir -p ./dist
[root@localhost swmm-toolkit]# python3 -m pip install -r build-requirements.txt
Requirement already satisfied: setuptools==65.5.1 in /usr/local/lib/python3.9/site-packages (from -r build-requirements.txt (line 4)) (65.5.1)
Requirement already satisfied: wheel==0.38.1 in /usr/local/lib/python3.9/site-packages (from -r build-requirements.txt (line 5)) (0.38.1)
Requirement already satisfied: scikit-build==0.11.1 in /usr/local/lib/python3.9/site-packages (from -r build-requirements.txt (line 6)) (0.11.1)
Requirement already satisfied: cmake==3.21 in /usr/local/lib/python3.9/site-packages (from -r build-requirements.txt (line 7)) (3.21.0)
Requirement already satisfied: swig==4.0.2 in /usr/local/lib/python3.9/site-packages (from -r build-requirements.txt (line 8)) (4.0.2)
Requirement already satisfied: packaging in /usr/local/lib/python3.9/site-packages (from scikit-build==0.11.1->-r build-requirements.txt (line 6)) (23.1)
Requirement already satisfied: distro in /usr/local/lib/python3.9/site-packages (from scikit-build==0.11.1->-r build-requirements.txt (line 6)) (1.8.0)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
[root@localhost swmm-toolkit]# python3 setup.py bdist_wheel
/usr/local/lib/python3.9/site-packages/setuptools/_distutils/dist.py:264: UserWarning: Unknown distribution option: 'cmake_process_manifest_hook'
  warnings.warn(msg)


--------------------------------------------------------------------------------
-- Trying "Unix Makefiles" generator
--------------------------------
---------------------------
----------------------
-----------------
------------
-------
--
Not searching for unused variables given on the command line.
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is GNU 7.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is GNU 7.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit/_cmake_test_compile/build
--
-------
------------
-----------------
----------------------
---------------------------
--------------------------------
-- Trying "Unix Makefiles" generator - success
--------------------------------------------------------------------------------

Configuring Project
  Working directory:
    /usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit/_skbuild/linux-aarch64-3.9/cmake-build
  Command:
    cmake /usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit -G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit/_skbuild/linux-aarch64-3.9/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/usr/local/bin/python3 -DPYTHON_VERSION_STRING:STRING=3.9.4 -DPYTHON_INCLUDE_DIR:PATH=/usr/local/include/python3.9 -DPYTHON_LIBRARY:FILEPATH=/usr/local/lib/libpython3.9.a -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/usr/local/lib/python3.9/site-packages/skbuild/resources/cmake -DCMAKE_BUILD_TYPE:STRING=Release

-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_C: -fopenmp (found version "4.5") 
fatal: Not a git repository (or any parent directory)：.git
CMake Error at swmm-solver/CMakeLists.txt:121 (install):
  install EXPORT given no DESTINATION!


CMake Error at swmm-solver/CMakeLists.txt:130 (install):
  install FILES given no DESTINATION!


-- Configuring incomplete, errors occurred!
See also "/usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit/_skbuild/linux-aarch64-3.9/cmake-build/CMakeFiles/CMakeOutput.log".
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/skbuild/setuptools_wrap.py", line 582, in setup
    env = cmkr.configure(cmake_args,
  File "/usr/local/lib/python3.9/site-packages/skbuild/cmaker.py", line 229, in configure
    raise SKBuildError(

An error occurred while configuring with CMake.
  Command:
    cmake /usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit -G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit/_skbuild/linux-aarch64-3.9/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/usr/local/bin/python3 -DPYTHON_VERSION_STRING:STRING=3.9.4 -DPYTHON_INCLUDE_DIR:PATH=/usr/local/include/python3.9 -DPYTHON_LIBRARY:FILEPATH=/usr/local/lib/libpython3.9.a -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/usr/local/lib/python3.9/site-packages/skbuild/resources/cmake -DCMAKE_BUILD_TYPE:STRING=Release
  Source directory:
    /usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit
  Working directory:
    /usr/local/lib/python3.9/site-packages/swmm-python-0.14.0/swmm-python-0.14.0/swmm-toolkit/_skbuild/linux-aarch64-3.9/cmake-build
Please see CMake's output for more information.
[root@localhost swmm-toolkit]#

**Comment by bemcdonnell** [comment]
@Lijifeng1234 I’ll preface my idea that I’ve been wrong many times before. This really seems like a directory problem. Can you send a screenshot of your directory and show where all the code and project contents are?  SWMM-toolkit needs SWMM-solver in a very specific layout. I just want to confirm the sub modules are correct. Then I want to make sure the right versions are checked out of SWMM and SWMM-toolkit

**Comment by Lijifeng1234** [comment]
> @Lijifeng1234 I’ll preface my idea that I’ve been wrong many times before. This really seems like a directory problem. Can you send a screenshot of your directory and show where all the code and project contents are? SWMM-toolkit needs SWMM-solver in a very specific layout. I just want to confirm the sub modules are correct. Then I want to make sure the right versions are checked out of SWMM and SWMM-toolkit
There is too much content to show it all. But all the files and content are from the above two links, which you can download and view. When testing your version separately, find that swmm-solver is incomplete with extern, and then supplement it with Stormwater-Management-Model.
![image](https://user-images.githubusercontent.com/42561978/236618316-152bbe61-9d22-4c90-bab8-0a52e0685ee0.png)

![image](https://user-images.githubusercontent.com/42561978/236618300-1d234056-b58b-458b-b9a0-4caf65aa3b5c.png)

**Comment by bemcdonnell** [comment]
@abhiramm7 any thoughts with this one?

**Comment by bemcdonnell** [comment]
I’m converting this to a discussion since we have not opted to support amd64 at this time.

**Comment by abhiramm7** [comment]
@Lijifeng1234 can you share the commands you ran? We should be able to run amd without any issues. We can help you build it :)

**Comment by karosc** [comment]
swmm-toolkit has supported aarch since [version 0.15.1](https://pypi.org/project/swmm-toolkit/0.15.1/) musl linux support coming soon!
